contracts/truefi3/interface/IDebtInstrument.sol

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.6.10;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {IERC721} from "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+
+/**
+ * @title IDebtInstrument
+ * @notice Represents debt in the TrueFi protocol
+ * @dev This is the basic interface for representing Debt in TrueFi
+ * This interface is meant to be extended to support different debt types
+ *
+ * An NFT represents some debt where:
+ *  - Each debt has an owner and a borrower
+ *  - The owner mints and funds a loan
+ *  - The borrower can withdraw funds from the loan
+ *  - The borrower calls repay() to pay back the loan
+ *  - Repayments are withdrawable by the owner
+ *  - Every DebtInstrument has an underlying token
+ *  - Only principal debt is tracked
+ *
+ **/
+interface IDebtInstrument is IERC721 {
+    /// @dev The owner address lends tokens and withdraws repayments
+    function owner() external view returns (address);
+
+    /// @dev The borrower address borrows and repays the loan
+    function borrower() external view returns (address);
+
+    /// @dev The underlying token denominating the debt
+    function token() external view returns (address);
+
+    /// @dev Principal debt amount
+    function principal() external view returns (uint256);
+
+    /// @dev Each debt has an expiration timestamp
+    function expiry() external view returns (uint256);
+
+    /// @dev Borrower calls this function to borrow `amount` of tokens
+    function borrow(uint256 amount) external;
+
```

Comment:
Is there a use case where we would want to fund a DebtInstrument but not transfer the principal to the borrower?

If not, then I would prefer to delete this function and just have `fund()` transfer directly from owner to borrower while recording the amount in `principal`. Saves an extra transaction and an extra ERC20 transfer.

Reply:
If `fund()` is restricted to only be callable by the NFT's owner (probably what we currently call agencies?), then we can put a `borrow()` function in the owner contract that delegates to `DI.fund()`.

Reply:
Yes, I think there is a case for using fund(). I think funds should be stored in the DebtInstrument until they are borrowed since it allows flexibility in the design on the debtInstrument (ongoing funding/borrowing).